File: sistema-de-suporte/src/app/services/chamado.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Chamado {
  id: number;
  titulo: string;
  descricao: string;
  status: string;
  dataAbertura: string;         // string no formato YYYY-MM-DD
  dataFechamento?: string;      // opcional, string ou undefined
  setor?: string;               // opcional
}

@Injectable({
  providedIn: 'root'
})
export class ChamadoService {

  private apiUrl = 'https://localhost:7063/api/chamados';

  constructor(private http: HttpClient) {}

  getChamados(): Observable<Chamado[]> {
    return this.http.get<Chamado[]>(this.apiUrl);
  }

  getChamado(id: number): Observable<Chamado> {
    return this.http.get<Chamado>(`${this.apiUrl}/${id}`);
  }

  criarChamado(chamado: Chamado): Observable<Chamado> {
    return this.http.post<Chamado>(this.apiUrl, chamado);
  }

  atualizarChamado(id: number, chamado: Chamado): Observable<Chamado> {
    return this.http.put<Chamado>(`${this.apiUrl}/${id}`, chamado);
  }

  excluirChamado(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: SuporteTecnico/SuporteTecnico/SuporteTecnico.Dominio/Chamado.cs
namespace SuporteTecnico.Dominio
{
    public class Chamado
    {
        public int Id { get; set; }
        public string Titulo { get; set; } = string.Empty;
        public string Descricao { get; set; } = string.Empty;
        public string Status { get; set; } = "Aberto";
        public DateTime DataAbertura { get; set; } = DateTime.UtcNow;
        public DateTime? DataFechamento { get; set; }

        // Enum Setor
        public Setor Setor { get; set; }
    }

    public enum Setor
    {
        Coordenação,
        Secretaria,
        Direção,
        RH
    }
}
